### PR TITLE
BUG: Fix osna idxmax over labels without any accumulated time

### LINKED ISSUE #380: BUG: argmax of empty sequence when using OSNA method for location identification
I get an error when running the location identification method on Geolife data. The error trace is the following:
```---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
/var/folders/m9/1q3cs2cx43qd5c82ljgh9g3w0000gr/T/ipykernel_47137/2164060891.py in <module>
----> 1 sp_w_purpose = ti.analysis.location_identifier(sp_test, method="OSNA")

~/MIE/trackintel/trackintel/analysis/location_identification.py in location_identifier(staypoints, method, pre_filter, **pre_filter_kwargs)
     68         method_val = freq_method(sp[f], "home", "work")
     69     elif method == "OSNA":
---> 70         method_val = osna_method(sp[f])
     71     else:
     72         raise ValueError(f"Method {method} does not exist.")

~/MIE/trackintel/trackintel/analysis/location_identification.py in osna_method(staypoints)
    319         sp_pivot.loc[redo_work["work"], "purpose"] = "home"
    320         sp_pivot.loc[redo_work["work"], "work"] = pd.NaT
--> 321         sp_idxmax_work = sp_pivot.groupby(["user_id"])["work"].idxmax()
    322         sp_pivot.loc[sp_idxmax_work.dropna(), "purpose"] = "work"
    323 

~/miniconda3/envs/gis/lib/python3.9/site-packages/pandas/core/groupby/groupby.py in wrapper(*args, **kwargs)
    945                 return self.apply(curried)
    946 
--> 947             return self._python_apply_general(curried, self._obj_with_exclusions)
    948 
    949         wrapper.__name__ = name

~/miniconda3/envs/gis/lib/python3.9/site-packages/pandas/core/groupby/groupby.py in _python_apply_general(self, f, data)
   1285             data after applying f
   1286         """
-> 1287         keys, values, mutated = self.grouper.apply(f, data, self.axis)
   1288 
   1289         return self._wrap_applied_output(

~/miniconda3/envs/gis/lib/python3.9/site-packages/pandas/core/groupby/ops.py in apply(self, f, data, axis)
    818             # group might be modified
    819             group_axes = group.axes
--> 820             res = f(group)
    821             if not _is_indexed_like(res, group_axes, axis):
    822                 mutated = True

~/miniconda3/envs/gis/lib/python3.9/site-packages/pandas/core/groupby/groupby.py in curried(x)
    934 
    935             def curried(x):
--> 936                 return f(x, *args, **kwargs)
    937 
    938             # preserve the name so we can detect it when calling plot methods,

~/miniconda3/envs/gis/lib/python3.9/site-packages/pandas/core/series.py in idxmax(self, axis, skipna, *args, **kwargs)
   2352         nan
   2353         """
-> 2354         i = self.argmax(axis, skipna, *args, **kwargs)
   2355         if i == -1:
   2356             return np.nan

~/miniconda3/envs/gis/lib/python3.9/site-packages/pandas/core/base.py in argmax(self, axis, skipna, *args, **kwargs)
    641                 return -1
    642             else:
--> 643                 return delegate.argmax()
    644         else:
    645             # error: Incompatible return value type (got "Union[int, ndarray]", expected

~/miniconda3/envs/gis/lib/python3.9/site-packages/pandas/core/arrays/_mixins.py in argmax(self, axis, skipna)
    134         if not skipna and self.isna().any():
    135             raise NotImplementedError
--> 136         return nargminmax(self, "argmax", axis=axis)
    137 
    138     def unique(self: NDArrayBackedExtensionArrayT) -> NDArrayBackedExtensionArrayT:

~/miniconda3/envs/gis/lib/python3.9/site-packages/pandas/core/sorting.py in nargminmax(values, method, axis)
    445         return func(values, axis=axis)
    446 
--> 447     return _nanargminmax(values, mask, func)
    448 
    449 

~/miniconda3/envs/gis/lib/python3.9/site-packages/pandas/core/sorting.py in _nanargminmax(values, mask, func)
    456     non_nan_idx = idx[~mask]
    457 
--> 458     return non_nan_idx[func(non_nans)]
    459 
    460 

~/miniconda3/envs/gis/lib/python3.9/site-packages/numpy/core/overrides.py in argmax(*args, **kwargs)

~/miniconda3/envs/gis/lib/python3.9/site-packages/numpy/core/fromnumeric.py in argmax(a, axis, out, keepdims)
   1214     """
   1215     kwds = {'keepdims': keepdims} if keepdims is not np._NoValue else {}
-> 1216     return _wrapfunc(a, 'argmax', axis=axis, out=out, **kwds)
   1217 
   1218 

~/miniconda3/envs/gis/lib/python3.9/site-packages/numpy/core/fromnumeric.py in _wrapfunc(obj, method, *args, **kwds)
     55 
     56     try:
---> 57         return bound(*args, **kwds)
     58     except TypeError:
     59         # A TypeError occurs if the object does have such a method in its

ValueError: attempt to get argmax of an empty sequence
```

The error seems to come from a few cases where it applies the `idxmax()` function on a dataframe where one row only contains NaNs. It's not entirely clear to me however why this happens. An example csv is attached, you can test it with the following lines:
```
sp_test = ti.io.read_staypoints_csv("test_loc_identify.csv")
sp_w_purpose = ti.analysis.location_identifier(sp_test, method="OSNA")
```

[test_loc_identify.csv](https://github.com/mie-lab/trackintel/files/8248831/test_loc_identify.csv)

### PR DESCRIPTION
**Pull request by bifbof**
With removing pandas version <=v.1.2.5 we got also a change that broke
the osna method in location identification.
I could not really trace the origin of the change but in the end the
`idxmax` from pandas uses `nanargmax` from numpy. That works even if
everything is nan, but only if it is nan and not any other NA value.
In our case that was pd.NaT for the Timedeltas. In my opinion that is a bug
in pandas as their documentation is stating that it works for NA
(that includes NaT).
Well in the end I only had to change the dtype of the aggregated column. :D

TLDR: Changed dtype such that idxmax gets np.nan

closes #380 
(and partially #378)

### CODE REVIEW

**Review by NinaWie — APPROVED** [review]
Looks good to me, thanks for fixing!
